src/script/lua_api/l_object.cpp

```diff
@@ -2200,6 +2200,64 @@ int ObjectRef::l_get_day_night_ratio(lua_State *L)
 	return 1;
 }
 
+// set_minimap_modes(self, modes, modeindex)
+int ObjectRef::l_set_minimap_modes(lua_State *L)
+{
+	NO_MAP_LOCK_REQUIRED;
+
+	// Arg 1 should be a player
+	ObjectRef *ref = checkobject(L, 1);
+	RemotePlayer *player = getplayer(ref);
+	if (player == NULL)
+		return 0;
+
+	// Arg 2 should be a table (of tables)
+	if (!lua_istable(L, 2)) {
+		return 0;
+	}
+
+	// Arg 3 should be a number
+	s16 wantedmode = lua_tonumber(L, 3);
+
+	std::vector<MinimapMode> modes;
+
+	lua_pushnil(L);
+	while (lua_next(L, 2) != 0) {
+		/* key is at index -2, value is at index -1 */
+		if (lua_istable(L, -1)) {
+			MinimapMode mode;
+			std::string type = getstringfield_default(L, -1, "type", "");
+			if (type == "off")
+				mode.type = 0;
+			else if (type == "surface")
+				mode.type = 1;
+			else if (type == "radar")
+				mode.type = 2;
+			else if (type == "texture") {
+				mode.type = 3;
+				mode.texture = getstringfield_default(L, -1, "texture", "");
+				mode.scale = getintfield_default(L, -1, "scale", 1);
+			} else {
+				mode.type = -1;
+			}
+
+			if (mode.type >= 0) {
```

Comment:
this will result in unknown modes causing the selected mode index to be incorrect. Not sure how best to handle this

Reply:
Actually it will ignore mode addition, issuing a warning (code being updated). Another way to handle that could be to issue an error on Lua side.